templates/redis.socketed.template.yml

```
@@ -0,0 +1,84 @@
+run:
+  - file:
+      path: /etc/service/redis/run
+      chmod: "+x"
+      contents: |
+        #!/bin/sh
+        exec 2>&1
+        exec thpoff chpst -u redis -U redis /usr/bin/redis-server /etc/redis/redis.conf
+  - file:
+      path: /etc/service/redis/log/run
+      chmod: "+x"
+      contents: |
+        #!/bin/sh
+        mkdir -p /var/log/redis
+        exec svlogd /var/log/redis
+  - file:
+      path: /etc/runit/3.d/10-redis
+      chmod: "+x"
+      contents: |
+        #!/bin/bash
+        sv stop redis
+  - replace:
+      filename: "/etc/redis/redis.conf"
+      from: "port 6379"
+      to: "port 0"
+  - replace:
+      filename: "/etc/redis/redis.conf"
+      from: "# unixsocket /tmp/redis.sock"
+      to: "unixsocket /shared/tmp/redis.sock"
```

`/shared/tmp` is only writable by `discourse` user by default, so the UNIX socket needs to be created elsewhere:
```suggestion
      to: "unixsocket /shared/redis_data/redis.sock"
```
To align with PostgreSQL, `redis_run` could be added as new directory for the socket.